modules/servodyn/src/BladedInterface.f90

```diff
@@ -221,10 +221,11 @@ SUBROUTINE CallBladedLegacyDLL ( u, filt_fromSCglob, filt_fromSC, p, dll_data, E
    aviFAIL = 0                ! bjj, this won't necessarially work if aviFAIL is INTENT(OUT) in DLL_Procedure()--could be undefined???
 
       !Convert to C-type characters: the "C_NULL_CHAR" converts the Fortran string to a C-type string (i.e., adds //CHAR(0) to the end)
+      ! Note: the optional size is specified to avoid an array mismatch issue with some intel compilers in debug (gfortran doesn't notice)
 
-   avcOUTNAME = TRANSFER( TRIM(dll_data%RootName)//C_NULL_CHAR,   avcOUTNAME )
-   accINFILE  = TRANSFER( TRIM(dll_data%DLL_InFile)//C_NULL_CHAR, accINFILE  )
-   avcMSG     = TRANSFER( C_NULL_CHAR,                            avcMSG     ) !bjj this is intent(out), so we shouldn't have to do this, but, to be safe...
+   avcOUTNAME  = TRANSFER( TRIM(dll_data%RootName)//C_NULL_CHAR,   avcOUTNAME, p%avcOUTNAME_LEN )
+   accINFILE   = TRANSFER( TRIM(dll_data%DLL_InFile)//C_NULL_CHAR, accINFILE,  LEN_TRIM(dll_data%DLL_InFile)+1 )
+   avcMSG      = TRANSFER( C_NULL_CHAR,                            avcMSG,     LEN(ErrMsg)+1 ) !bjj this is intent(out), so we shouldn't have to do this, but, to be safe...
```

Comment:
For consistency, wouldn't you want the size argument on avcMSG to be 1? The other 2 lines appear to be using the size of the first argument, not the size of the second.

Reply:
I believe that argument should be the length of the variable on the lhs:

```
size
    (Input; optional) Must be scalar and of type integer. It provides the number of elements for the output result.
```

from 
https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/language-reference/a-to-z-reference/t-to-z/transfer.html

Reply:
Yeah, maybe you're right. I had been looking at the part that says `Its size is the smallest that is possible to hold all of source`... of course the documentation also says the size argument is optional. 

I can see where the `avcMSG` and `avcOUTNAME` lines might cause issues without that size argument, but the `accINFILE` line has the same size/length values for the first and second arguments so that is somewhat puzzling.

There are a lot of `TRANSFER` statements elsewhere in the code, too. I recall changing many of those statements years ago so they do not use the TRIM function... I think it was causing seg faults in the Simulink interface. It would be interesting to know if those TRANSFER statements also need the SIZE argument or if the way they are implemented works with your build.

Reply:
I think the compiler didn't like the `avcOUTNAME` and `avcMSG`.  Then I changed the `accINFILE` at the same time without first checking if it was an issue.

I've only seen issues with this while running the code compiled with the Intel Debug compile (2021.4.0 on linux).